DTS-STN/DECD-Design-System
# Packaging improvements and cleanup

In trying to use your package in the Eligibility Estimator project, I've been making some improvements as I go.

Here are the changes I've made:
- run `npm install` using recent npm version (update `package-lock.json` to new structure)
  - Are you guys using an outdated npm version? I'm surprised I had to do this. See "breaking changes" below for more info.
- remove unused dependency `joi`
  - The only reference to `joi` is in commented code, so it is unused.
- remove unneeded dependency `user`
  - No idea why this was added in the first place, if you look at its source code it literally does nothing. [npm page](https://www.npmjs.com/package/user)
- add `prepack` script, so you just need to do `npm pack` and the build will run automatically
  - Documentation is [here](https://docs.npmjs.com/cli/v8/using-npm/scripts), pretty straightforward. No need to `build` before `pack`.
- move most dependencies to devDependencies, set react and prop-types as regular dependencies, update rollup config accordingly
  - For whatever reason your deps were scattered around regular/peer/dev. Dependencies not required to run the package should be under devDependencies. As well, dependencies required to run the package like `react` are regular dependencies, and not peerDependencies. So, I moved everything over accordingly. This drastically reduces the number of packages installed in projects that use your package, see [here](https://github.com/DTS-STN/eligibility-estimator/pull/232/commits/edc32479bb2072460e55bf5d99cf437897261127) for how much better the Eligibility Estimator is now!
- add .gitattributes for consistent line-endings cross-platform
  - This is mainly a Windows thing, but it tells Git to use LF endings, which is what you are using already. 
- add `cross-env` to test script, for cross-platform usage
  - You had done this already for the `build` script, I just applied it to `test` as well.

Please let me know if there are any commits or changes I should remove.

---

**Reminder: Add all changes for this PR to the CHANGELOG.md**
(should I do this?)

Main Reviewer: ?

#### Description & Background Context:
Packaging improvements and cleanup. Should have very little functional impact.

#### Where should the reviewer start?
Review each commit individually. This will give you a better idea of what changed and why, rather than looking at all the changes at once.

#### How should this be tested?
Run all your automated builds and tests, see if anything has exploded. I ran `test`, `build`, `storybook`, `build-storybook`, and even pulled a component into EE, and it seems to still work fine. 

#### Are there any breaking changes?
No, at least not on purpose. Things to note are:
- if you have automated scripts that previously did `npm run build` then `npm pack`, those should be simplified to `npm pack`
- you may need to update your local `npm` version, as `package-lock.json` has been migrated to the latest structure. For reference, I'm using `8.5.3`.
